### DISCUSSION TITLE
Why is Chrome running for development so slow?

### DISCUSSION THREAD

**Discussion by MortensenC** [extension-js/extension.js#74] [General]
I'm developing an extension to add functionality to a web and opening that web on this Chrome is so slow.

**Comment by nangerno** [comment]
Generally, Chrome Driver can be used on development chrome and extension related to customize the default functionalities of Chrome.
So I think that it can reset all setting items on running Chrome on your development steps.
But it is not a big problem. If  you once deploy your extension, it can be work well done not slow.

**Reply by MortensenC** [reply]
Sorry, I don't get you; I'll try to be as clear as possible:
Once I run **npm run dev**, chrome opens fast, but when I try to open the website, it takes about 1 minute and a half or more to open it, and then I get an error on the terminal:
<img width="893" alt="image" src="https://github.com/cezaraugusto/extension.js/assets/435573/8b08b55b-39b4-4688-8b7a-a21ebb7f1df8">

I cannot wait for 1 minute and a half every time I change something on my extension; if you can guide me on where I can look to improve this, I can put my hands on this package. 
Despite that, this package is great, thank you very much, I would like to contribute.

**Reply by nangerno** [reply]
I think that this problem is due to this extension you developed. To fix this problem, I'd like to know what function of this extension.
And on your image, what is the meaning of "EMFILE: too many open files".
Maybe it's caused by this issue?

**Reply by MortensenC** [reply]
But my extension is not doing anything yet; I'm adding a button just for testing. The problem is opening WhatsApp on that Chrome.

**manifest**:
```
 "permissions": [
    "activeTab",
    "webNavigation",
    "storage"
  ],
  "host_permissions": [
    "https://web.whatsapp.com/*"
  ],
  "background": {
    "service_worker": "src/background.js"
  },
  "content_scripts": [
    {
      "matches": ["https://web.whatsapp.com/*"],
      "js": ["src/scripts.js"]
    }
  ]
```

```
// background.js
chrome.webNavigation.onCompleted.addListener(function(details) {
    console.log('Navigation complete: ', details);
  }, {url: [{urlMatches: 'https://web.whatsapp.com/'}]});
```
  
```
// content.js
function addCustomButton() {
  const button = document.createElement('button');
  button.textContent = 'Personalized button';
  button.style.position = 'fixed';
  button.style.top = '10px';
  button.style.right = '10px';
  button.style.zIndex = '1000';
  button.addEventListener('click', () => {
    alert('¡Clicked button!');
  });
  document.body.appendChild(button);
}

window.addEventListener('load', () => {
  addCustomButton();
});
```

**Reply by nangerno** [reply]
Generally, when listening for events or performing operations with current page(in you case to open whatsapp), delay can be occoured.
If you replace and rewrite add text to the current page instead of button click event, you can fix this problem.
So first, please test with this method.

**Reply by MortensenC** [reply]
I think that this is not the case. I replaced all the code inside **addCustomButton** with a console.log, and it's still so slow. Then I decided to remove all the extensions from that Chrome, and it was still working slowly. Then I decided to try with other pages, like a newspaper https://www.infobae.com/ or whatever I try to open, and it's still so slow to load.
Did you try using the Chrome you open when you run **npm run dev**? Is it working well for you?

And just FYI, is not my Mac, I have a Apple M3 Pro with 36 GB of RAM

**Reply by nangerno** [reply]
Of course, I used it and it worked well. There is only one thing left....
Please run your code on the other PC.

**Reply by MortensenC** [reply]
Yes, you were right. I tried it on my Intel Mac, and it works perfectly. I'll have to see what's happening with this Mac with an M chip.
In this Mac, I have an error with the watchers... I'm trying to solve that, maybe the problem is there.

Many thanks for your time!

**Reply by cezaraugusto** [reply]
hi @MortensenC which errors are you seeing? did you got able to run the project?

**Reply by MortensenC** [reply]
Hi @cezaraugusto , yes I was able to run it with out errors, I had to delete node_modules and install everything again, maybe I used an old node version, it would be awesome to add the correct node version to the package json and force to use that one. But WhatsApp web is still very slow on that chrome.
